src/screens/TopicPageScreen/index.js

```diff
@@ -450,13 +462,27 @@ const TopicPageScreen = (props) => {
           contentInsetAdjustmentBehavior={feeds?.length > 1 ? 'automatic' : 'never'}
         />
       </View>
-      <ButtonAddPostTopic topicName={topicName} onRefresh={onRefresh} />
+      <ButtonAddPostTopic topicName={topicName} onRefresh={onRefresh} followType={followType} />
       <BlockComponent
         ref={refBlockComponent}
         refresh={onBlockCompleted}
         refreshAnonymous={onDeleteBlockedPostCompleted}
         screen="topic_screen"
       />
+      <BottomSheetFollow
+        ref={bottomSheetFollowRef}
+        onClose={() => bottomSheetFollowRef.current.close()}
+        topicName={topicDetail.name}
+        memberCount={memberCount}
+        setMemberCount={setMemberCount}
+        isFollow={isFollow}
+        setIsFollow={setIsFollow}
+        followType={followType}
+        setFollowType={setFollowType}
+        getTopicDetail={getTopicDetail}
+        username={user?.myProfile.username}
+        profilePicture={user?.myProfile.profile_pic_path}
+      />
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

New components `ButtonAddPostTopic`, `BlockComponent`, and `BottomSheetFollow` have been added with new props related to following topics. Make sure these new props are properly defined and used within these components.
```diff
+ <ButtonAddPostTopic topicName={topicName} onRefresh={onRefresh} followType={followType} />
+ <BlockComponent
+   ref={refBlockComponent}
+   refresh={onBlockCompleted}
+   refreshAnonymous={onDeleteBlockedPostCompleted}
+   screen="topic_screen"
+ />
+ <BottomSheetFollow
+   ref={bottomSheetFollowRef}
+   onClose={() => bottomSheetFollowRef.current.close()}
+   topicName={topicDetail.name}
+   memberCount={memberCount}
+   setMemberCount={setMemberCount}
+   isFollow={isFollow}
+   setIsFollow={setIsFollow}
+   followType={followType}
+   setFollowType={setFollowType}
+   getTopicDetail={getTopicDetail}
+   username={user?.myProfile.username}
+   profilePicture={user?.myProfile.profile_pic_path}
+ />
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->